src/components/shared/GoogleMaps/GoogleMaps.tsx

```diff
@@ -40,71 +29,65 @@ const circleOptions = {
   strokeWeight: 2,
 };
 
-class GoogleMaps extends React.Component<Props, State> {
-  readonly state = {
-    lat: this.props.lat,
-    lng: this.props.lng,
-    loading: false,
-    error: false,
-  }
-
-  componentDidMount() {
-    if ((this.state.lat === undefined || this.state.lng === undefined) && this.props.address) {
-      this._updateCoordinates();
-    }
-  }
+function GoogleMaps(props: Props) {
+  const [coordinates, setCoordinates] = useState<google.maps.LatLngLiteral>({ lat: 0, lng: 0 });
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
+  const [isMounted, setIsMounted] = useState<boolean>(true);
 
-  componentDidUpdate = (prevProps: Props) => {
-    if (this.props.address && (prevProps.address !== this.props.address)) {
-      this._updateCoordinates();
-    }
-  }
+  const debouncedAddress = useDebounce(props.address, 500);
 
-  render() {
-    const { lat, lng } = this.state;
-    if (this.state.loading) {
-      return <AudioLoading height={48} width={96} />;
-    }
-    
-    if (this.state.error) {
-      return <h1>Error</h1>;
-    }
+  useEffect(() => {
+    setLoading(true);
+    setIsMounted(true);
 
-    if (lat === undefined || lng === undefined) {
-      return <h1>Please provide a valid address</h1>;
-    }
+    fetchCoordinates(debouncedAddress || '')
+      .then(({ lat, lng }: google.maps.LatLngLiteral) => {
+        if (!isMounted) return;
 
-    return (
-      <GoogleMap defaultZoom={13.5} defaultCenter={{ lat, lng }}>
-        {this.props.showCircle && <Circle center={{ lat, lng }} options={circleOptions} radius={900} />}
-        {this.props.showMarker && <Marker position={{ lat, lng }} />}
-      </GoogleMap>
-    );
-  }
-
-  _debouncedFetchCoordinates = debounce(() => {
-    fetchCoordinates(this.props.address || '').then(({ lat, lng }) => {
-      this.props.getCoordinates && this.props.getCoordinates({ lat, lng });
-      this.setState({
-        lat,
-        lng,
-        error: false,
-        loading: false,
-      })
-    }).catch(error => {
-      console.error(error);
-      this.setState({
-        error: true,
-        loading: false,
+        props.getCoordinates && props.getCoordinates({ lat, lng });
+        setCoordinates({ lat, lng });
+        setLoading(false);
+        setError(false);
       })
-    });
-  }, 1000);
+      .catch(error => {
+        if (!isMounted) return;
+
+        console.error(error);
+        setLoading(false);
+        setError(true);
+      });
+      
+    return cleanUp;
+  },
+    [debouncedAddress]
+  );
```

Comment:
Indentation here got a little confusing for me; think this might be clearer as:

    }, [debouncedAddress]);

...so that it goes back to (and closes) the indentation level of `useEffect` on a single line

Reply:
prettifier agrees! good catch